linode_api4/linode_client.py

```diff
@@ -1433,6 +1433,620 @@ def cancel(self):
+        """
+        Returns a boolean value indicating if this bucket has a corresponding
+        TLS/SSL certificate that was uploaded by an Account user.
+
+        API Documentation: https://www.linode.com/docs/api/object-storage/#object-storage-tlsssl-cert-view
+
+        :param cluster_id: The ID of the cluster this bucket exists in.
+        :type cluster_id: str
+
+        :param bucket: The bucket name.
+        :type bucket: str
+
+        :returns: A boolean indicating if this Bucket has a corresponding
+                  TLS/SSL certificate that was uploaded by an Account user.
+        :rtype: boolean
+        """
+        result = self.client.get(
+            "/object-storage/buckets/{}/{}/ssl".format(cluster_id, bucket)
+        )
+
+        if "errors" in result:
+            raise UnexpectedResponseError(
+                "Unexpected response when getting the TLS/SSL certs indicator of a bucket!",
+                json=result,
+            )
+
+        return MappedObject(**result)
+
+    def ssl_cert_upload(self, cluster_id, bucket, certificate, private_key):
+        """
+        Upload a TLS/SSL certificate and private key to be served when you
+        visit your Object Storage bucket via HTTPS. Your TLS/SSL certificate and
+        private key are stored encrypted at rest.
+
+        To replace an expired certificate, delete your current certificate and
+        upload a new one.
+
+        API Documentation: https://www.linode.com/docs/api/object-storage/#object-storage-tlsssl-cert-upload
+
+        :param cluster_id: The ID of the cluster this bucket exists in.
+        :type cluster_id: str
+
+        :param bucket: The bucket name.
+        :type bucket: str
+
+        :param certificate: Your Base64 encoded and PEM formatted SSL certificate.
+                            Line breaks must be represented as “\n” in the string
+                            for requests (but not when using the Linode CLI)
+        :type certificate: str
+
+        :param private_key: The private key associated with this TLS/SSL certificate.
+                            Line breaks must be represented as “\n” in the string
+                            for requests (but not when using the Linode CLI)
+        :type private_key: str
+
+        :returns: A boolean indicating if this Bucket has a corresponding
+                  TLS/SSL certificate that was uploaded by an Account user.
+        :rtype: boolean
+        """
+        params = {
+            "certificate": certificate,
+            "private_key": private_key,
+        }
+        result = self.client.post(
+            "/object-storage/buckets/{}/{}/ssl".format(cluster_id, bucket),
+            data=params,
+        )
+
+        if "errors" in result:
+            raise UnexpectedResponseError(
+                "Unexpected response when uploading TLS/SSL certs!",
+                json=result,
+            )
+
+        return MappedObject(**result)
```

Comment:
Could we move the implementation of these methods to their corresponding classes? 
e.g.
`bucket_access_modify(...)` would become `bucket.access_modify(...)`, where bucket is an arbitrary ObjectStorageBucket object.